tests/jeos/check_efi_setup.pm

```diff
@@ -0,0 +1,271 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Check EFI boot in JeOS images
+#   1) Get gpt drive and find ESP
+#   2) Check ESP partition alignment
+#   3) Perform fsck on ESP
+#   4) Check EFI settings after firstboot
+#   5) Reinstall grub2, disable secureboot, reboot
+#   6) Check EFI settings after grub2 reinstall
+#   7) Reinstall shim, enable secureboot, reboot
+#   8) Check EFI settings after shim reinstall
+#   9) Reinitialize bootloader using pbl
+#   10) Check EFI settings after pbl
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use Mojo::Base 'opensusebasetest';
+use testapi;
+use utils qw(zypper_call is_efi_boot);
+use version_utils qw(is_leap is_opensuse is_sle);
+use Utils::Backends qw(is_hyperv is_svirt_except_s390x);
+use jeos qw(reboot_image set_grub_gfxmode);
+use constant {
+    SYSFS_EFI_BITS      => '/sys/firmware/efi/fw_platform_size',
+    FSTAB               => '/etc/fstab',
+    GRUB_DEFAULT        => '/etc/default/grub',
+    GRUB_CFG            => '/boot/grub2/grub.cfg',
+    SYSCONFIG_BOOTLADER => '/etc/sysconfig/bootloader'
+};
+use Data::Dumper;
+
+sub get_expected_efi_settings {
+    my $settings      = {};
+    my $efi_rec_label = (is_opensuse() ? (lc get_var('DISTRI')) : 'sles');
+    my $efi_exec      = $efi_rec_label . '/grubx64.efi';
+
+    if (!get_var('DISABLE_SECUREBOOT', 0)) {
+        $efi_exec = $efi_rec_label . '/shim.efi';
+        $efi_rec_label .= '-secureboot';
+    }
+    $settings->{exec}  = '/EFI/' . $efi_exec;
+    $settings->{label} = $efi_rec_label;
+
+    return $settings;
+}
+sub efibootmgr_current_boot {
+    my $ebm_raw = script_output 'efibootmgr --verbose';
+    my $h       = {};
+
+    ($h->{bootid}) = $ebm_raw =~ /^BootCurrent:\s+(\d+)/m;
+    die("Missing BootCurrent: in efibootmgr output") unless (exists($h->{bootid}) or defined($h->{bootid}));
+
+    if ($ebm_raw =~ /^(Boot$h->{bootid}\*\s+(.+)\s+(\S+\([^\)]+\)\/?)+.*|Boot$h->{bootid}\*\s+(.+))$/m) {
+        $h->{label} = $2 // $4;
+    }
+    die("Missing lable in Boot$h->{bootid} entry") unless (exists($h->{label}) or defined($h->{label}));
+
+    ($h->{exec}) = $ebm_raw =~ /^Boot$h->{bootid}\*.*File\(([^\)]+)\).*/m;
+    if (exists($h->{exec}) && defined($h->{exec})) {
+        $h->{exec} =~ s'\\'/'g;
+    } else {
+        $h->{exec} = undef;
+    }
+
+    return $h;
+}
+
+sub check_efi_state {
+    is_efi_boot or die "Image did not boot in UEFI mode!\n";
+    my $expected = shift;
+
+    # check UEFI firmware bitness
+    validate_script_output "cat ${ \SYSFS_EFI_BITS }", sub { $_ == 64 };
```

Comment:
Maybe this, because of constants
```suggestion
    validate_script_output('cat ' . SYSFS_EFI_BITS, sub { $_ == 64 };
```

Reply:
Usual dispute between concatenation vs interpolation in our PRs. :D
I am fine with both, personally I feel we (as test contributors) prefer to use interpolation over concatenation. Does concatenation fit you more ?

Reply:
I agree, looks like tabs vs spaces :). And I wouldn't say anything if it would look normal like `"cat $SYSFS_EFI_BITS"` or `"cat ${SYSFS_EFI_BITS}"`. But duo to `constant` we need this `${ \SYSFS_EFI_BITS }` and only this looks like concatenation fits more here. But I don't have hard feelings. Just do as you want.